docs/content/concepts/configuration/advanced-config-types.mdx

```diff
@@ -0,0 +1,260 @@
+            "pet_stats": ConfigWithUnion(
+                pet=Cat(meows=10),
+            )
+        }
+    ),
+)
+```
+
+### Enum types
+
+Python enums which subclass `Enum` are supported as config fields. Here, we define a schema that takes in a list of users, whose roles are specified as enum values:
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_enum_schema_config endbefore=end_enum_schema_config dedent=4
+from dagster import Config, RunConfig, op, job
+from enum import Enum
+
+class UserPermissions(Enum):
+    GUEST = "guest"
+    MEMBER = "member"
+    ADMIN = "admin"
+
+class ProcessUsersConfig(Config):
+    users_list: Dict[str, UserPermissions]
+
+@op
+def process_users(config: ProcessUsersConfig):
+    for user, permission in config.users_list.items():
+        if permission == UserPermissions.ADMIN:
+            print(f"{user} is an admin")  # noqa: T201
+
+@job
+def process_users_job():
+    process_users()
+
+op_result = process_users_job.execute_in_process(
+    run_config=RunConfig(
+        {
+            "process_users": ProcessUsersConfig(
+                users_list={
+                    "Bob": UserPermissions.GUEST,
+                    "Alice": UserPermissions.ADMIN,
+                }
+            )
+        }
+    ),
+)
+```
+
+### Validated config fields
+
+Config fields can have custom validation logic applied using [Pydantic validators](https://docs.pydantic.dev/usage/validators/). Pydantic validators are defined as methods on the config class, and are decorated with the `@validator` decorator. Here, we define some validators on a configured user's name and username, which will throw exceptions if incorrect values are passed in the launchpad or from a schedule or sensor.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_validated_schema_config endbefore=end_validated_schema_config dedent=4
+from dagster import Config, RunConfig, op, job
+from pydantic import validator
+
+class UserConfig(Config):
+    name: str
+    username: str
+
+    @validator("name")
+    def name_must_contain_space(cls, v):
+        if " " not in v:
+            raise ValueError("must contain a space")
+        return v.title()
+
+    @validator("username")
+    def username_alphanumeric(cls, v):
+        assert v.isalnum(), "must be alphanumeric"
+        return v
+
+executed = {}
+
+@op
+def greet_user(config: UserConfig) -> None:
+    print(f"Hello {config.name}!")  # noqa: T201
+    executed["greet_user"] = True
+
+@job
+def greet_user_job() -> None:
+    greet_user()
+
+# Input is valid, so this will work
+op_result = greet_user_job.execute_in_process(
+    run_config=RunConfig(
+        {"greet_user": UserConfig(name="Alice Smith", username="alice123")}
+    ),
+)
+
+# Name has no space, so this will fail
+op_result = greet_user_job.execute_in_process(
+    run_config=RunConfig(
+        {"greet_user": UserConfig(name="John", username="johndoe44")}
+    ),
+)
+```
```

Comment:
This may have to be in a follow-up PR so the link tests pass, but I think adding a **Related** section at the end of this page would be a good idea. It could contain links to the other configuration documentation.